CONTRIBUTING.md

```diff
@@ -188,11 +188,58 @@ with one another are rolled up.
 Speaking of tests, Rust has a comprehensive test suite. More information about
 it can be found [here][rctd].
 
-### External Dependencies
+### External Dependencies (subrepo)
 
-Currently building Rust will also build the following external projects:
+As a developer to this repository, you don't have to treat the following external projects
+differently from other crates that are directly in this repo:
 
 * [clippy](https://github.com/rust-lang/rust-clippy)
+
+They are just regular files and directories. This is in contrast to `submodule` dependencies.
+
+If you want to synchronize or otherwise work with subrepos, install the `git subrepo` command via
+instructions found at https://github.com/ingydotnet/git-subrepo
+
+#### Creating a new subrepo dependency
+
+If you want to create a new subrepo dependency from an existing repository, call (from this
+repository's root directory!!)
+
+```
+git subrepo clone https://github.com/rust-lang/rust-clippy.git src/tools/clippy
+```
+
+This will create a new commit, which you may not rebase under any circumstances! Delete the commit
+and redo the operation if you need to rebase.
+
+Now you're done, the `src/tools/clippy` directory behaves as if clippy were part of the rustc
+monorepo, so no one but you (or others that synchronize subrepos) needs to have `git subrepo`
+installed.
+
+#### Synchronizing a subrepo
+
+There are two synchronization directions: push and pull.
+
+A `git subrepo push src/tools/clippy`
+takes all the changes that
+happened to the copy in this repo and creates commits on the remote repo that match the local
+changes (so ever local commit that touched the subrepo causes a commit on the remote repo). Again,
+Even this `push` operation creates a commit in this repo that you need to get merged without
+rebasing. This is very important in order to make future synchronizations work.
```

Comment:
So the idea is, I will `git subrepo push`, and then make a rustc PR (likely self-r+ p=1) and a tool PR? And then what, I wait until the rustc PR lands before landing the tool PR, or vice versa? Either way there is a chance of conflicts where one PR lands and the other one does not, and it is unclear how bad that is. We should give some general guidance to at least have a consistent process.

Reply:
We have no experience here. Not sure how to build guidance without really knowing how it will behave in practice.

Reply:
Come up with a best guess, adjust it later if it doesn't work? ;)